### DISCUSSION TITLE
Unused parameters

### DISCUSSION THREAD

**Discussion by vectro** [spotbugs/spotbugs#3089] [Ideas]
Is there a reason we don’t have a check for unused parameters? In my experience this is a huge reg flag that something has been missed — often there is something that a method should be taking into account, but isn’t. 

Given how far into the weeds many of the SpotBugs checks are, I guess the absence of an unused parameter check must be intentional. But what is the motivation?

**Comment by ThrawnCA** [comment]
Is it not covered by unused variable checks?

**Comment by vectro** [comment]
I don't think so and did not observe this in my testing.

I am not a Java expert but it appears to me that SpotBugs checks for various types of "dead local store"; however unused parameters are never stored.

The closest thing we have is IP_PARAMETER_IS_DEAD_BUT_OVERWRITTEN but that only checks for parameters that are written (thus, not really unused).

**Comment by ThrawnCA** [comment]
If a rule is added for this, it should probably ignore any methods that implement interfaces, since they are more likely to have a legitimate reason to ignore parameters (the interface declares that info will be supplied, but a specific implementation may not need it).

**Reply by vectro** [reply]
We should probably exclude anything marked `override` or which directly provides an implemented interface, or which overrides an abstract method when defining a non-abstract class, I.e., all the situations where we can be confident that this method does not define its own API. 

However I’m less confident that we should suppress this error when overriding an overloaded method, since it’s possible that we’re overriding the wrong method.

**Reply by fonsdant** [reply]
It would possible to use unnamed variables in implementations, wouldn't it?

**Reply by vectro** [reply]
@fonsdant, I’m not sure I understand what you mean by that. Could you show a trivial example?

**Reply by fonsdant** [reply]
```java
public interface SomeInterface {
    void someMethod(int someArg1, int someArg2);
}

public class SomeImplementation1 implements SomeInterface {
    @Override
    void someMethod(int someArg1, int _) {
        // ...
    }
}

public class SomeImplementation2 implements SomeInterface {
    @Override
    void someMethod(int _, int someArg2) {
        // ...
    }
}
```
@vectro, this way we let the dev decides whether the rule should be applied or not to the implementations.

**Reply by vectro** [reply]
I see. Unfortunately this is only available starting in Java 22.
